jpo-ode-core/src/main/java/us/dot/its/jpo/ode/wrapper/serdes/MessagingDeserializer.java

```diff
@@ -1,43 +1,45 @@
 /*******************************************************************************
- * Copyright 2018 572682
- * 
- * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * Copyright 2018 572682.
+ *
+ * <p>Licensed under the Apache License, Version 2.0 (the "License"); you may not
  * use this file except in compliance with the License.  You may obtain a copy
- * of the License at
- * 
- *   http://www.apache.org/licenses/LICENSE-2.0
- * 
- * Unless required by applicable law or agreed to in writing, software
+ * of the License at</p>
+ *
+ *   <p>http://www.apache.org/licenses/LICENSE-2.0</p>
+ *
+ * <p>Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
  * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
  * License for the specific language governing permissions and limitations under
- * the License.
+ * the License.</p>
  ******************************************************************************/
-package us.dot.its.jpo.ode.wrapper.serdes;
 
-import java.util.Map;
+package us.dot.its.jpo.ode.wrapper.serdes;
 
 import org.apache.kafka.common.serialization.Deserializer;
-
 import us.dot.its.jpo.ode.util.SerializationUtils;
 
-public class MessagingDeserializer<T> implements Deserializer<T> {
-
-    SerializationUtils<T> deserializer = new SerializationUtils<T>();
-
-    @Override
-    public void configure(Map<String, ?> configs, boolean isKey) {
-        // nothing to do
-    }
-
-    @Override
-    public T deserialize(String topic, byte[] data) {
-        return deserializer.deserialize(data);
-    }
-
-    @Override
-    public void close() {
-        // nothing to do
-    }
-
+/**
+ * MessagingDeserializer is a generic base class implementing the Kafka Deserializer interface to
+ * provide serialization of objects for use in Kafka messages.
+ *
+ * <p>This class uses a generic type parameter, allowing it to handle serialization
+ * of various types. Internal serialization is performed using an instance of the SerializationUtils
+ * class, which leverages Kryo for efficient object serialization.</p>
+ *
+ * <p>The class is declared as sealed, restricting which other classes can directly extend it. It
+ * will
+ * soon be marked as final to prevent incorrect usage through unnecessary subtyping</p>
+ *
+ * @param <T> the type of data to be serialized
+ */

```

Comment:
(blocking) These comments indicate that the class is for serialization, but the class is for deserialization.

Reply:
corrected in https://github.com/CDOT-CV/jpo-ode/pull/131/commits/d7f2bbecede1d0da5d3a38ea7db8fce0985373ee